comms.go

```diff
@@ -209,12 +210,14 @@ func (d *Device) WriteRawImageToButton(btnIndex int, rawImg image.Image) error {
 
 func (d *Device) rawWriteToButton(btnIndex int, rawImage []byte) error {
 	// Based on set_key_image from https://github.com/abcminiuser/python-elgato-streamdeck/blob/master/src/StreamDeck/Devices/StreamDeckXL.py#L151
+
+	if Min(Max(btnIndex, 0), int(d.deviceType.numberOfButtons)) != btnIndex {
+		return errors.New(fmt.Sprintf("Invalid key index: %d", btnIndex))
+	}
```

Comment:
Simple error ported from the python library. Probably not necessary, but it won't hurt anything either.